src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs

```diff
@@ -78,6 +78,23 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
         .WithOpenApi("Handles application queries", string.Empty, "application_get")
         .RequireAuthorization()
         .WithParameterValidation();
+
+    endpointRouteBuilder.MapDelete("/api/draftApplications/{id}", async Task<Results<Ok<CancelDraftApplicationResponse>, BadRequest<ProblemDetails>>> (string id, HttpContext ctx, CancellationToken ct, IMessageBus messageBus) =>
+       {
+         var userId = ctx.User.GetUserContext()?.UserId;
+
+         bool IdIsNotGuid = !Guid.TryParse(id, out _); if (IdIsNotGuid)
+         {
+           return TypedResults.BadRequest(new ProblemDetails() { Title = "ApplicationId is not valid" });
+         }
+
+         var cancelledApplicationId = await messageBus.InvokeAsync<string>(new CancelDraftApplicationCommand(id, userId!), ct);
+
+         return TypedResults.Ok(new CancelDraftApplicationResponse(cancelledApplicationId));
+       })
+       .WithOpenApi("Cancel a draft application for the current user", "Changes status to cancelled", "draftapplication_delete")
```

Comment:
I kept this as delete to better match for the frontend since we are going to use .delete. I could change this to cancel as well.